Feature/ce high data

AlvaroGRuiz opened the pull request:
OP#759

OP#1540

AlvaroGRuiz committed:
[REL] 🏗️  Check point

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[IMP] ✨ All logic and schema of form is ready to use. (Translations are missing.)

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

juan-pe commented on energy_communities_crm/services/crm_lead_service.py:
```diff
@@ -53,6 +66,33 @@ def create(self, **params):
+                return metadata_line.get("value") == "yes"
+        return False
+
+    def _check_coordinator_found(self, params):
```
Hi Alvaro!
Here are some thoughts about this method:

line 79 is suitable to raise exceptions, can you encapsulate it between a `try...except block?

what do you think if we move this pattern to a constant in the file?
something like:
```python
import re

DISCARD_NAMES_RE = re.compile(r"\[(ON-HOLD|TO-DELETE)\]?i")

# in _check_coordinator_found

DISCARD_NAMES_RE.check(company.name)
```
Regex are always hard to compute, so it's better if we calculate it before. Also, look that I added a "?i" at the end that made this regex case-insensitive. You cannot trust that those values will always be in upper-case.

About the if condition. It's hard to understand the meaning of that condition, I ask myself, in which case we add the coordinator_id to the metadata? Or why are we adding the coordinator_id to the metadata? The answers of those questions are the meaning of the condition, so lets to specify it. Something like:
```python

can_append_coordinator = and([company is not None, company.hierarchy_level == "coordinator", not re.search(pattern, company.name)])
if can_append_coordinator:
    params["metadata"].append(
        {"key": "coordinator_id", "value": company.id}
￼   )
```

More things, you name it as `_check_coordinator_found`, when I read this name, I will expect if coordinator is on parameters, without any actions. In this method, apart from check if coordinator is in params, we add it to the metadata, I propose or change the name of the method to represent what is doing (`_add_coordinator_to_metadata` for example), or change the logic of the method to adjust to the name.

juan-pe reviewed (changes requested):
Hi @AlvaroGRuiz! thanks a lot for your work! looks nice!

only two points to review
* I think function `_check_coordinator_know_value` can be improved, I have written you a comment about it in the code
* Can you write a little test to check that the method `create` is working as you expect?

If you have any troubles with that, we can do a paring session and solve it! 👐🏽 😄 

Thanks a lot!

AlvaroGRuiz committed:
[FIX] 🐛 Fix onchange on form, add try cath and change name of function

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[IMP] ✨ Implement test and preformace code

Author: Alvaro Garcia <alvaro.garcia@somit.coop>
Author: Juan Pedro Sánchez <juanpe.sanchez@somenergia.coop>

AlvaroGRuiz committed:
[IMP] ✨ Apply change on Form WebSite and i18n translate

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[I18N] 🌐 Fix translate in EU

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[I18N] 🌐 Fix i18n on form view crm creation

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[I18N] 🌐 Fix i18n

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[I18N] 🌐 Fix i18n

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[IMP] ✨ New mapping metadata in EasyCreationCompanyWizard

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix error for last data

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[IMP] ✨ Check coordinator company

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[IMP] ✨ Improvement of the multicompany easy creation wizard

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[I18N] 🌐 New translate

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix disable field on form crm_lead

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix EasyCreationCompanyWizard. Maping ce_member_status and ce_status. Save lang on partner. Fix tax for TF.

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix check https://

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix reference of taxes

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

daniquilez committed:
[IMP] ✨ mcec move res_partner_bank and bank journal creation to service_invoicing

Co-authored-by: Alvaro Garcia <alvaro.garcia@somit.coop>

daniquilez committed:
[TEST] ✅ Unify MultiCompanyEasyCreation testing

daniquilez committed:
[IMP] ✨ Rename subscription journal and hide bank_id from mcec wizard

daniquilez committed:
[IMP] ✨ Setup default canary taxes on MCEC

daniquilez committed:
[IMP] ✨ MCEC on change legal form update chart of accounts considering canary

daniquilez committed:
[FIX] 🐛 Remove white space on tax template ref

AlvaroGRuiz committed:
[I18N] 🌐 Translate Neighborhood community

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

daniquilez committed:
[IMP] ✨ Draft of including canary on demo data

AlvaroGRuiz committed:
[IMP] ✨ Implements new tests in energy_communitites_services_invoices for canary cases

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz commented on energy_communities_crm/demo/crm_lead_demo.xml:
```diff
@@ -137,154 +137,154 @@
+      (0, 0, {'key': 'ce_number_of_members', 'value': '96'}),
+      (0, 0, {'key': 'ce_creation_date', 'value': '1994-09-01'}),
+      (0, 0, {'key': 'contact_phone', 'value': '666666666'}),
+      (0, 0, {'key': 'email_from', 'value': 'community3@bar.fake'}),
```
Replace this
      (0, 0, {'key': 'email_from', 'value': 'community3@bar.fake'}),
for this
      (0, 0, {'key': 'email_from', 'value': 'community6@bar.fake'}),

AlvaroGRuiz committed:
[FIX] 🐛 Data demo replace (0, 0, {'key': 'email_from', 'value': 'community3@bar.fake'}) for this (0, 0, {'key': 'email_from', 'value': 'community6@bar.fake'})

Author: Alvaro Garcia <alvaro.garcia@somit.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix all tests on energy_communities_service_invoicing

Author: Alvaro Garcia <alvaro.garcia@somit.coop>
Author: Dani Quilez <dani.quilez@somenergia.coop>

AlvaroGRuiz committed:
[FIX] 🐛 Fix assetion for selfconsumption journal and typo on set_new_distribution_table method

Author: Alvaro Garcia <alvaro.garcia@somit.coop>
Author: Dani Quilez <dani.quilez@somenergia.coop>

daniquilez committed:
[IMP] ✨ Dynamic change creation taxes for canary

daniquilez committed:
Merge branch 'feature/ce_high_data' of github.com:somcomunitats-coop/odoo-ec into feature/ce_high_data

daniquilez committed:
[IMP] ✨ Use existing bank journal as first journal to be used on res_partner_pank creation

daniquilez committed:
[REL] 🏗️ Release 2025-11-03

Bump energy_communities version to 16.0.0.7.0
Bump energy_communities_cooperator version to 16.0.0.4.1
Bump energy_communities_crm version to 16.0.0.3.0
Bump energy_communities_service_invoicing version to 16.0.0.7.0
Bump energy_selfconsumption version to 16.0.0.5.12

 Co-authored-by: Alvaro Garcia <alvaro.garcia@somit.coop>
 Co-authored-by: Juanpe Sánchez  <juanpe.sanchez@somenergia.coop>

daniquilez reviewed (approved):
All tested and fine.

juan-pe reviewed (approved):
👍🏽 great job!! 🤩 🤩
